### PR TITLE
Retain position of extension function calls

### LINKED ISSUE #11523: Semanticdb: occurence range for extension method application isn't correct
## Compiler version

3.0.0-RC1

## Minimized code

```Scala
extension (s: String)
  def foo: Int = 42

val a = "asd".foo
```

## Output

```
src/main/scala/Main.scala
-------------------------

Summary:
Schema => SemanticDB v4
Uri => src/main/scala/Main.scala
Text => empty
Language => Scala
Symbols => 4 entries
Occurrences => 7 entries

Symbols:
_empty_/Main$package. => final package object _empty_
_empty_/Main$package.a. => val method a
_empty_/Main$package.foo(). => method foo
_empty_/Main$package.foo().(s) => param s

Occurrences:
[0:0..0:0) <= _empty_/Main$package.
[0:11..0:12) <= _empty_/Main$package.foo().(s)
[0:14..0:20) => scala/Predef.String#
[1:6..1:9) <= _empty_/Main$package.foo().
[1:11..1:14) => scala/Int#
[3:4..3:5) <= _empty_/Main$package.a.
[3:8..3:8) => _empty_/Main$package.foo().
```

## Expectation

```
src/main/scala/Main.scala
-------------------------

Summary:
Schema => SemanticDB v4
Uri => src/main/scala/Main.scala
Text => empty
Language => Scala
Symbols => 4 entries
Occurrences => 7 entries

Symbols:
_empty_/Main$package. => final package object _empty_
_empty_/Main$package.a. => val method a
_empty_/Main$package.foo(). => method foo
_empty_/Main$package.foo().(s) => param s

Occurrences:
[0:0..0:0) <= _empty_/Main$package.
[0:11..0:12) <= _empty_/Main$package.foo().(s)
[0:14..0:20) => scala/Predef.String#
[1:6..1:9) <= _empty_/Main$package.foo().
[1:11..1:14) => scala/Int#
[3:4..3:5) <= _empty_/Main$package.a.
[3:15..3:17) => _empty_/Main$package.foo().
```

The problem is in the last occurrence range.
Instead of `[3:8..3:8) ` it should be [3:15..3:17)

### PR DESCRIPTION
**Pull request by Kordyjan**
Fixes #11523.

### CODE REVIEW

**Review by bishabosha — CHANGES_REQUESTED** [review]
Thanks for taking this on!

I tried with right associative operator extension applications, which still have a zero length span, e.g.
```scala
package ext

extension (s: String)
  def :*: (i: Int): (String, Int) = (s, i)

val b = "foo" :*: 23
```
yields
```scala
expect/Extension.scala
----------------------

Summary:
Schema => SemanticDB v4
Uri => Extension.scala
Text => empty
Language => Scala
Symbols => 5 entries
Occurrences => 14 entries

Symbols:
ext/Extension$package. => final package object ext
ext/Extension$package.`:*:`(). => method :*:
ext/Extension$package.`:*:`().(i) => param i
ext/Extension$package.`:*:`().(s) => param s
ext/Extension$package.b. => val method b

Occurrences:
[0:8..0:11): ext <- ext/
[2:0..2:0): <- ext/Extension$package.
[2:11..2:12): s <- ext/Extension$package.`:*:`().(s)
[2:14..2:20): String -> scala/Predef.String#
[3:6..3:9): :*: <- ext/Extension$package.`:*:`().
[3:11..3:12): i <- ext/Extension$package.`:*:`().(i)
[3:14..3:17): Int -> scala/Int#
[3:21..3:27): String -> scala/Predef.String#
[3:29..3:32): Int -> scala/Int#
[3:37..3:37): -> scala/Tuple2.apply().
[3:37..3:38): s -> ext/Extension$package.`:*:`().(s)
[3:40..3:41): i -> ext/Extension$package.`:*:`().(i)
[5:4..5:5): b <- ext/Extension$package.b.
[5:14..5:14): -> ext/Extension$package.`:*:`(). // Note: zero length span here
```
we should probably watch out for possible interaction with the `selectSpan` method: https://github.com/lampepfl/dotty/blob/4ed026cb7c0e3e1f97456172e382f199c1aea8da/compiler/src/dotty/tools/dotc/semanticdb/ExtractSemanticDB.scala#L515

**Comment by Kordyjan** [comment]
I solved the problem with inline operators by adding an attachment containing the original span of the operator. I don't think there is any nicer way of passing this information for synthesized selects.

File: compiler/src/dotty/tools/dotc/ast/Trees.scala [hunk 1]
```diff
@@ -415,6 +415,15 @@ object Trees {
         qualifier.typeOpt.member(name).atSignature(Signature.NotAMethod, name)
       case _ =>
         super.denot
+
+    def nameSpan(using Context): Span =
```
Comment (bishabosha) [inline compiler/src/dotty/tools/dotc/ast/Trees.scala:419]: I tried this which did not need any attachments (I don't think its that important to do the point span for synthetic):
```scala
def nameSpan(using Context): Span =
  if span.exists then
    val point = span.point
    if name.toTermName == nme.ERROR then Span(point)
    else if qualifier.span.start > span.start then // right associative
      val realName = name.stripModuleClassSuffix.lastPart
      Span(span.start, span.start + realName.length, point)
    else
      Span(point, span.end, point)
  else span
```

**Review by bishabosha — APPROVED** [review]
LGTM, I have found an optimisation without attachments

**Pull request merged**